### PR TITLE
feat: Publish multi-arch docker image (amd64 and arm64)

### PR DESCRIPTION
**Pull request by project0**
## What is the problem I am trying to address?

We want to switch our workload to arm64 (AWS Graviton) for improved cost-efficiency. One of the remaining images not proving arm64 compatible images is athens.

This PR supersede #1794 as the pipeline was moved to github actions

## How is the fix applied?

extend github actions workflow

## What GitHub issue(s) does this PR fix or close?

Fixes #

### CODE REVIEW

**Comment by DrPsychick** [comment]
@project0 We'll take a look at this after the 0.12.0 version is released.

**Comment by mloiseleur** [comment]
Hello @DrPsychick ,

 0.12.0 version has been released. Is the path clear to review this PR, now ?

**Comment by DrPsychick** [comment]
Hey @mloiseleur, generally yes, we're on track now again the latest Athens.
Next up will be 0.12.1 with bugfixes (soon). I will take a look at this within the next 2 weeks.

**Review by DrPsychick — CHANGES_REQUESTED** [review]

File: cmd/proxy/Dockerfile [hunk 1]
```diff
@@ -8,15 +8,24 @@
 ARG GOLANG_VERSION=1.20
 ARG ALPINE_VERSION=3.17
 
-FROM golang:${GOLANG_VERSION}-alpine AS builder
+FROM --platform=$BUILDPLATFORM golang:${GOLANG_VERSION}-alpine AS builder
+
+ARG VERSION="unset" \
+		TARGETARCH
```
[the remark this answers is not in the document]
Comment (project0) [reply cmd/proxy/Dockerfile:14]: No, someone kicked of the file with tabs, but i can change format to white spaces

**Review by DrPsychick — CHANGES_REQUESTED** [review]
How did you test this?

I cannot get it to run through on my fork, so please test it in your fork so that the actions actually run through.

Also, one minor thing: all other workflows use `.yml` extension, so keep it consistent please.

https://github.com/DrPsychick/athens/actions/runs/4869859787/jobs/8684877269

**Comment by project0** [comment]
@DrPsychick i am sorry, i should have tested it better. There seem to be some issues with re-usables here, so i decided to avoid them at the end.

**Review by DrPsychick — CHANGES_REQUESTED** [review]
minor fix and we're good I think.

**Comment by justenwalker** [comment]
Hey @project0 ; thanks for implementing this!

Perhaps we could also get arm64 builds in the .goreleaser.yml as well?

https://github.com/gomods/athens/pull/1870/files#diff-42e26dc67aed8aa3edb2472b4403288c1699fb6dc47419b9a475f0f224fe4689

**Review by DrPsychick — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**